Surface_mesh_approximation/doc/Surface_mesh_approximation/NamedParameters.txt

```diff
@@ -0,0 +1,173 @@
+// number_of_iterations: number of relaxation iterations after initialization
+// anchors: output anchor points
+// triangles: output triplets of indexed triangles
+
+CGAL::Surface_mesh_approximation::approximate_triangle_mesh(tm,
+  CGAL::parameters::seeding_method(method).
+    max_number_of_proxies(number_of_proxies).
+    number_of_iterations(number_of_iterations).
+    anchors(std::back_inserter(anchors)).
+    triangles(std::back_inserter(triangles)));
+\endcode
+
+\cgalHeading{List of Available Named Parameters}
+
+In this package, all functions optional parameters are implemented as BGL optional named parameters and listed below.
+
+In the following, we assume that the following types are provided as template parameters of surface mesh approximation functions and classes. Note that the type is more specific for some of these functions.
+
+<ul>
+<li>`TriangleMesh` implements a `FaceListGraph`
+<li>`GeomTraits` a geometric traits class in which constructions are performed and predicates evaluated. Throughout this package, a \cgal `Kernel` fulfills the requirements.
+</ul>
+
+The named parameters available in this package are categorized into 3 groups below.
+
+\cgalHeading{1. Parameters for the Approximation Step}
+\cgalNPTableBegin
+
+\cgalNPBegin{geom_traits} \anchor VSA_geom_traits
+ the geometric traits instance in which the mesh approximation operation should be performed.\n
+\b Type: a Geometric traits class.\n
+\b Default type is
+\code
+typename CGAL::Kernel_traits<
+  typename boost::property_traits<
+    typename boost::property_map<TriangleMesh, CGAL::vertex_point_t>::type>::value_type>::Kernel
+\endcode
+\cgalNPEnd
+
+\cgalNPBegin{vertex_point_map} \anchor VSA_vertex_point_map
+ is the property map with the points associated to the vertices of the polygon mesh `pmesh`.\n
+\b Type: a class model of `ReadablePropertyMap` with
+`boost::graph_traits<TriangleMesh>::%vertex_descriptor` as key type and
+`GeomTraits::Point_3` as value type. \n
+\b Default value is \code boost::get(CGAL::vertex_point, pmesh)\endcode
+\cgalNPEnd
+
+\cgalNPBegin{verbose_level} \anchor VSA_verbose_level
+set the verbose level of the function.\n
+\b Type : \c CGAL::Surface_mesh_approximation::Verbose_level \n
+\b Default value is `CGAL::Surface_mesh_approximation::SILENT`
+\cgalNPEnd
+
+\cgalNPBegin{seeding_method} \anchor VSA_seeding_method
+the selection of seeding method.\n
+\b Type : \c CGAL::Surface_mesh_approximation::Seeding_method \n
+\b Default value is `CGAL::Surface_mesh_approximation::Seeding_method::HIERARCHICAL`
+\cgalNPEnd
+
+\cgalNPBegin{max_number_of_proxies} \anchor VSA_max_number_of_proxies
+the maximum number of proxies used to approximate the input mesh.\n
+\b Type : `boost::optional<std::size_t>` \n
+\b Default value is `boost::none`
+\cgalNPEnd
+
+\cgalNPBegin{min_error_drop} \anchor VSA_min_error_drop
+the minimum total error drop to approximate the input mesh.\n
+\b Type : `boost::optional<GeomTraits::FT>` \n
```

Comment:
if there is a default value, it makes no sense to have an optional, it should be FT.

Reply:
fixed